server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionEnvironment.java

```diff
@@ -72,6 +72,17 @@ public interface GarbageCollectionEnvironment {
    */
   Stream<Reference> getReferences();
 
+  /**
+   * Return a list of all TableIDs in the
+   * {@link org.apache.accumulo.core.metadata.schema.Ample.DataLevel} for which we are considering
+   * deletes. When operating on DataLevel.USER this will return all user table ids. When operating
+   * on DataLevel.METADATA this will return the table id for the accumulo.metadata table. When
+   * operating on DataLevel.ROOT this will return the table id for the accumulo.root table.
+   *
+   * @return The table ids
+   */
+  Set<TableId> getCandidateTableIDs();
```

Comment:
The term "operating on" used throughout this javadoc is very confusing. What does it mean to be "operating on" a particular level? I think we could choose a better verb to understand what's going on. Maybe "when gathering file candidates for deletion for DataLevel.USER, this will return a list of user table IDs".

Reply:
Maybe its confusing because it goes more with `GCRun.getCandidateTableIDs()`.  Could do the following to make it less confusing.

* Move the current javadoc on the interface to `GCRun.getCandidateTableIDs()` and replace `when operating on` with `the current data level is`
* Update the javadoc on this interface to make it generic like : 'Return a list of all TableIDs that should be seen in {@link #getReferences()} at the current time.  Immediately after this method returns the information it produced may be out of date relative to {@link #getReferences()} ... {@see GCRun#getCandidateTableIDs}`

Reply:
Moved javadoc around as suggested in  1f8d37305789fa1540158e32b54da86a7af3ae87